perception/image_projection_based_fusion/include/image_projection_based_fusion/fusion_node.hpp

```diff
@@ -0,0 +1,111 @@
+// Copyright 2021 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IMAGE_PROJECTION_BASED_FUSION__FUSION_NODE_HPP_
+#define IMAGE_PROJECTION_BASED_FUSION__FUSION_NODE_HPP_
+
+#include <image_projection_based_fusion/debugger.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+#include <autoware_auto_perception_msgs/msg/detected_objects.hpp>
+#include <sensor_msgs/msg/camera_info.hpp>
+#include <tier4_perception_msgs/msg/detected_objects_with_feature.hpp>
+
+#include <message_filters/pass_through.h>
+#include <message_filters/subscriber.h>
+#include <message_filters/sync_policies/approximate_time.h>
+#include <message_filters/synchronizer.h>
+#include <tf2_ros/buffer.h>
+#include <tf2_ros/transform_listener.h>
+
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace image_projection_based_fusion
+{
+using autoware_auto_perception_msgs::msg::DetectedObject;
+using autoware_auto_perception_msgs::msg::DetectedObjects;
+using tier4_perception_msgs::msg::DetectedObjectsWithFeature;
+using tier4_perception_msgs::msg::DetectedObjectWithFeature;
+
+template <class Msg>
+class FusionNode : public rclcpp::Node
+{
+public:
+  explicit FusionNode(const std::string & node_name, const rclcpp::NodeOptions & options);
+
+protected:
+  void cameraInfoCallback(
+    const sensor_msgs::msg::CameraInfo::ConstSharedPtr input_camera_info_msg, const int camera_id);
+
+  void fusionCallback(
+    typename Msg::ConstSharedPtr input_msg,
+    DetectedObjectsWithFeature::ConstSharedPtr input_roi0_msg,
+    DetectedObjectsWithFeature::ConstSharedPtr input_roi1_msg,
+    DetectedObjectsWithFeature::ConstSharedPtr input_roi2_msg,
+    DetectedObjectsWithFeature::ConstSharedPtr input_roi3_msg,
+    DetectedObjectsWithFeature::ConstSharedPtr input_roi4_msg,
+    DetectedObjectsWithFeature::ConstSharedPtr input_roi5_msg,
+    DetectedObjectsWithFeature::ConstSharedPtr input_roi6_msg,
+    DetectedObjectsWithFeature::ConstSharedPtr input_roi7_msg);
+
+  virtual void preprocess();
+
+  virtual void fusionOnSingleImage(
```

Comment:
I want to start with a verb.

Reply:
changed at [3e58d25](https://github.com/autowarefoundation/autoware.universe/pull/548/commits/3e58d25ec2c03790d6a9938bc43741128f695a6a)

Reply:
@yukkysaito 
sorry, changed to [ad2e762](https://github.com/autowarefoundation/autoware.universe/pull/548/commits/ad2e762b9d0c5d636e304035526d64800d61febf)